src/snowflake/connector/cache.py

```diff
@@ -129,14 +139,12 @@ def keys(self) -> list[K]:
     def items(self) -> list[tuple[K, V]]:
         with self._lock:
             values: list[tuple[K, V]] = []
-            for k in list(self._cache.keys()):
-                try:
-                    # TODO: this function could be further optimized by removing
-                    #  the need here to call __getitem here and call se;f._cache.keys()
-                    v = self.__getitem(k, should_record_hits=False)
-                    values.append((k, v))
-                except KeyError:
-                    continue
+            for k, v in list(self._cache.items()):
+                if is_expired(v.expiry):
+                    self._expiration(k)
+                    self._delitem(k)
+                else:
+                    values.append((k, v.entry))
```

Comment:
No, unfortunately I cannot reuse `_getitem` as it may remove elements without us knowing whether it has hit an expiration, or not. In which case the file cache would want to call `load_if_should()` but not over and over again, only once.

I also couldn't overwrite that function in this child class because that would cause other functions (like `clear_expired_entries` to call the file version, which is not what we want.

Reply:
I'm contemplating over the following implementation. because just looking at the code, I think the effect is the same as what you write here:
```python
for k in self._cache.keys():
    try:
        values.append((k, self._get_item(k, should_record_hits=False))
    except KeyError:
        pass
```

Reply:
My bad, I thought this was in the `SFDictFileCache`, yes what you posted almost works.
```python
for k in list(self._cache.keys()):
    try:
        values.append((k, self._get_item(k, should_record_hits=False))
    except KeyError:
        pass
```
will do the trick. However; you cannot change a regular dictionary while iterating over it, so the `list` becomes necessary.